sh_scrapy/extension.py

```diff
@@ -20,14 +23,13 @@ def __init__(self, crawler):
         self.hsref = hsref.hsref
         if not self.hsref.enabled:
             raise NotConfigured
-
+        self.pipe_writer = pipe_writer
         self.crawler = crawler
-        self._write_item = self.hsref.job.items.write
+        self._write_item = self.pipe_writer.write_item
         # https://github.com/scrapy/scrapy/commit/c76190d491fca9f35b6758bdc06c34d77f5d9be9
         exporter_kwargs = {'binary': False} if not IS_PYTHON2 else {}
         with ignore_warnings(category=ScrapyDeprecationWarning):
             self.exporter = PythonItemExporter(**exporter_kwargs)
-        log.msg("HubStorage: writing items to %s" % self.hsref.job.items.url)
```

Comment:
I removed that when I thought I'll be able to remove hsref completely, it makes sense to return it back, no problem. I can modify the message to "Writing items to %s", not mentioning pipe, will it be enough for distinction?